### ISSUE TITLE
docs: JSON schema incorrect for (some?) array types

### ISSUE THREAD

**Issue by claytonrcarter** [phpactor/phpactor#2739]
Hi, I was just exploring some of the config settings available and noticed that my editor (Zed, using `json-language-server`) is showing some valid JSON as invalid. In particular, it's showing that the `indexer.exclude_patterns` key should take an object when it in fact (I think!) should take an array of strings:  
![Screenshot 2024-09-03 at 9 53 53 PM](https://github.com/user-attachments/assets/0ecb0642-736e-40e9-8ece-fb91125b39fe)

I believe that this is because the dumped schema (I'm running `2024.06.30.0`) includes this:
```json
        "indexer.exclude_patterns": {
            "description": "Glob patterns to exclude while indexing",
            "type": [
                "object"
            ],
            "default": [
                "\/vendor\/**\/Tests\/**\/*",
                "\/vendor\/**\/tests\/**\/*",
                "\/vendor\/composer\/**\/*"
            ]
        },
```
but I believe (according to [one of the examples on the JSON schema website](https://json-schema.org/learn/getting-started-step-by-step#define-optional-properties)) that it should perhaps be something like this instead:
```diff
        "indexer.exclude_patterns": {
            "description": "Glob patterns to exclude while indexing",
-           "type": [
-               "object"
-           ],
+          "type": "array",
+          "items": {
+             "type": "string"
+          },
            "default": [
                "\/vendor\/**\/Tests\/**\/*",
                "\/vendor\/**\/tests\/**\/*",
                "\/vendor\/composer\/**\/*"
            ]
        },
```

I don't know JSON Schema at all, so there may be other ways to make this change, but I can say that making the above change to the schema file makes `json-language-server` happy and the above error goes away. Furthermore, it will report `Incorrect type. Expected "string".` if any of the array values are not strings.

I see that this is affecting some other settings, like `indexer.include_patterns`, too.

From poking around, I can see that the schema type for `indexer.exclude_patterns` is [being set to `array`](https://github.com/phpactor/phpactor/blob/master/lib/Indexer/Extension/IndexerExtension.php#L116) but then it's [changed back to "object"](https://github.com/phpactor/phpactor/blob/master/lib/Extension/Configuration/Model/JsonSchemaBuilder.php#L72-L74) when it's dumped. 

A quick fix may be to change the schema type to `"string[]"`, which [would then (just an educated guess) be turned into `"array"`](https://github.com/phpactor/phpactor/blob/master/lib/Extension/Configuration/Model/JsonSchemaBuilder.php#L88-L90). That would be an improvement, and then the `"items": {"type": "string"}` part could be saved for later, maybe.

**Comment by dantleech** [comment]
yes, it's a limitation with the way we validate configuration "options".

https://github.com/phpactor/phpactor/blob/345ac709d5bcbebb82784a03198a19d75efd8e9c/lib/Indexer/Extension/IndexerExtension.php#L115

We define it as `array` - we could introduce support for `list` - this would *not* allow us to specify generic types (i.e. `list<string>`) but I guess we can use "mixed".
